CMakeLists.txt

```diff
@@ -9,12 +9,39 @@ Emscripten.cmake"
     )
 endif()
 
+# Extract the project name & version from package.json
+if(EXISTS "${CMAKE_CURRENT_SOURCE_DIR}/package.json")
+    file(READ "${CMAKE_CURRENT_SOURCE_DIR}/package.json" PACKAGE_JSON_CONTENT)
+else()
+    message(FATAL_ERROR "`package.json` not found in ${CMAKE_CURRENT_SOURCE_DIR}")
+endif()
+if(${PACKAGE_JSON_CONTENT} MATCHES "\"name\":[ ]*\"([^\"]+)\"")
+    set(CLP_FFI_JS_PROJECT_NAME
+            ${CMAKE_MATCH_1}
+            CACHE STRING
+            "The name of the project parsed from `package.json`."
+            FORCE
```

Comment:
Why do we need/want to make these cache variables?

Reply:
my understanding of CMake cache variables is that they are like configuration variables that can be overridden by `-DXXX` passed to the `cmake` command. 

with that said, realistically, in this project, those variables are not getting referenced anywhere else, so hardcoding those (project_name=clp-ffi-js; version=0.0.0) may also work just fine. (with that said, those variables might come in handy if we ever use Doxygen to generate docs for the C++ code / add any other tools that might need such project metadata. ) what do you think?

Reply:
and setting cache variables allow docstrings (though it's not like we can't write a comment instead

i guess they don't strictly need to be cache variables

Reply:
> but yeah why don't we directly change the metadata in package.json and let cmake sync from it? lol

Yep.

> with that said, realistically, in this project, those variables are not getting referenced anywhere else, so hardcoding those (project_name=clp-ffi-js; version=0.0.0) may also work just fine. (with that said, those variables might come in handy if we ever use Doxygen to generate docs for the C++ code / add any other tools that might need such project metadata. ) what do you think?

I think this point is kind of unrelated to the above and sounds more like it's questioning the motivation of the PR? Anyway, it's not a huge amount of effort to parse the project name and version out of package.json (and you've already done it), so I think we should keep it.

> i guess they don't strictly need to be cache variables

Yep.

Reply:
> I think this point is kind of unrelated to the above and sounds more like it's questioning the motivation of the PR?

yes

> I think we should keep it.

sure. sounds good